src/main/java/net/minecraftforge/client/LevelRendererHooks.java

```diff
@@ -0,0 +1,123 @@
+package net.minecraftforge.client;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.math.Matrix4f;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.LevelRenderer;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraftforge.client.event.RegisterLevelRendererHooksEvent;
+
+/**
+ * Renders registered hooks to render during their appropriate {@link Phase}.
+ * 
+ * @see RegisterLevelRendererHooksEvent
+ */
+public class LevelRendererHooks
+{
+    private static final Multimap<Phase, Consumer<RenderContext>> HOOKS = ArrayListMultimap.create();
+    private static final Map<RenderType, Phase> RENDER_TYPE_PHASES = new HashMap<>();
+    public static float partialTicks = 0.0F;
+
+    /**
+     * @see RegisterLevelRendererHooksEvent
+     */
+    public static void register(Phase phase, Consumer<RenderContext> hook)
+    {
+        HOOKS.put(phase, hook);
+    }
```

Comment:
Instead of having this be public API and potentially being called at the wrong time if you add a helper method that the javadocs says is internal (or actually if it is only ever called once during startup and not say every time a new world is joined that then errors if called a second time) that fires the `RegisterLevelRendererHooksEvent` and make the `RegisterLevelRendererHooksEvent` take a `Multimap<Phase, Consumer<RenderContext>>` as a parameter. Then the only way people will have to actually register to the map is by doing so at the proper time (in the event) as that is the only place that the `register(Phase, Consumer<RenderContext>)` will be exposed